staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -31,32 +31,35 @@ import (
 	"k8s.io/apiserver/pkg/storage/value"
 	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
 	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/kms"
 	"k8s.io/apiserver/pkg/storage/value/encrypt/secretbox"
 )
 
 const (
 	aesCBCTransformerPrefixV1    = "k8s:enc:aescbc:v1:"
 	aesGCMTransformerPrefixV1    = "k8s:enc:aesgcm:v1:"
 	secretboxTransformerPrefixV1 = "k8s:enc:secretbox:v1:"
+	gkmsTransformerPrefixV1      = "k8s:enc:gkms:v1:"
 )
 
-// GetTransformerOverrides returns the transformer overrides by reading and parsing the encryption provider configuration file
-func GetTransformerOverrides(filepath string) (map[schema.GroupResource]value.Transformer, error) {
+// GetTransformerOverrides returns the transformer overrides by reading and parsing the encryption provider configuration file.
+// It takes an optional kmsFactory as an argument, which is used when a KMS based encryption backend is used.
+func GetTransformerOverrides(filepath string, kmsFactory kms.Factory) (map[schema.GroupResource]value.Transformer, error) {
```

Comment:
I do not think kmsFactories should be passed into these functions. Instead, this should be factored as an interface that can be plumbed in when the server options are set up.